airflow/configuration.py

```diff
@@ -469,11 +469,22 @@ class AirflowConfigParser(ConfigParser):
     }
 
     def __init__(self, *args, **kwargs):
+        default_config = kwargs.pop('default_config', None)
         ConfigParser.__init__(self, *args, **kwargs)
-        self.read_string(parameterized_config(DEFAULT_CONFIG))
+        self.defaults = ConfigParser(*args, **kwargs)
+
+        AirflowConfigParser._read_string(
```

Comment:
We could collapse this down a little by requiring a default config be specified:

```python
def __init__(self, **args, default_config=None, **kwargs):
    super(AirflowConfigParser, self).__init__(*args, **kwargs)
    self.defaults = ConfigParser(*args, **kwargs)
    if default_config is not None:
        AirflowConfigParser._read_string(self, default_config)
```

and then, further on where `conf` is created:

```python
conf = AirflowConfigParser(default_config=parametrize_config(DEFAULT_CONFIG))
conf.read(AIRFLOW_CONFIG)
```

this has the added benefit of isolating the internal scope of the `AirflowConfigParser` class from the module scope making it easier to test.